### PR TITLE
fast sigmoid

### PR DESCRIPTION
**Pull request by nicholas-leonard**
This PR provides a faster implement of `torch.Tensor.sigmoid`. As outlined in  #1026, the rationale for this PR is that TensorFlow has a 1.77x speedup over Torch LSTMs on CPU using Float tensors. This PR brings us to par with TF LSTMs. 

The downside is that `torch.Tensor.sigmoid` will lose some precision. The rough approximation introduces about 0.2% error. The speedup seems to warrant the approximation because the throughput for a typical LSTM goes from 455 samples/sec to 910 samples/sec, which compares well to the 809 samples/sec for TF. For a smaller LSTM those numbers are respectively, 3087 to 3900 for Torch vs  4139 for TF.

An alternative could be to implement a separate THTensor_(fastsigmoid) for this rough approximation.

### CODE REVIEW

**Comment by soumith** [comment]
we used to have fast exponential approximations in TH for various things, and we phased them all out because of very subtle training issues for various use-cases. So I am going to reject this PR for the same reason.
However, I think @pavanky the wizard can easily speed this up without loss of precision.

**Comment by pavanky** [comment]
@soumith The easiest thing to try is to try and incorporate the float versions of the math functions instead of defaulting to double versions for everything. This does still lose a bit of accuracy to existing versions but it should have the expected precision for floating point numbers.

**Comment by soumith** [comment]
that does seem reasonable

**Comment by nicholas-leonard** [comment]
@soumith Wizard @pavanky is sending a PR with TH_sigmoidf that gives about the same speedup. Wait out.
